window/src/os/windows/window.rs

```diff
@@ -569,6 +584,10 @@ impl WindowInner {
                 })
                 .detach();
             }
+            promise::spawn::spawn(async move {
+                PostMessageW(hwnd, extra_constants::UM_APPEARANCE_CHANGED, 0, 0);
+            })
+            .detach();
```

Comment:
Similar to my comments on the focus state changes, I think this is a bit heavyweight to do when toggling fullscreen: it should remain nice and snappy.

What I suggest is removing this and adjusting `TermWindow::resize` to re-fetch the OS parameters if `self.window_state != window_state`.  I think that captures the intent of propagating the border change when full screen mode is toggled.

```suggestion
```